test/Operators/hybrid/2d.jl

```diff
@@ -188,74 +188,70 @@ end
 
         # only horizontal component of divergence
         hdiv = Operators.Divergence()
-        @. dh += hdiv(h * Geometry.UVector(1.0)) # add the two components for full divergence +=
+        @. dh += hdiv(h) # add the two components for full divergence +=
         Spaces.weighted_dss!(dh)
 
-        return divuf
+        return dh
     end
 
-    # A horizontal Gaussian blob, centered at (-x_0, -z_0),
-    # with `a` the height of the blob and σ the standard deviation
-    function h_blob(coords, x_0, z_0, a = 1.0, σ = 0.2)
-        f = map(coords) do coord
-            a * exp(-((coord.x + x_0)^2 + (coord.z + z_0)^2) / (2 * σ^2))
-        end
 
-        return f
+    # A horizontal Gaussian blob, centered at (x_0, z_0),
+    # with `a` the height of the blob and σ the standard deviation
+    function h_blob(x, z, x_0, z_0, a = 1.0, σ = 0.2)
+        a * exp(-((x - x_0)^2 + (z - z_0)^2) / (2 * σ^2))
     end
-
-    # Spatial derivative of a horizontal Gaussian blob, centered at (-x_0, -z_0),
+    # Spatial gradient of a horizontal Gaussian blob, centered at (x_0, z_0),
     # with `a` the height of the blob and σ the standard deviation
-    function div_h_blob(coords, x_0, z_0, a = 1.0, σ = 0.2)
-        div_uf = map(coords) do coord
-            -a * (exp(-((coord.x + x_0)^2 + (coord.z + z_0)^2) / (2 * σ^2))) /
-            (σ^2) * ((coord.x + x_0) + (coord.z + z_0))
-        end
+    ∇h_blob(x, z, x_0, z_0, a = 1.0, σ = 0.2) =
+        -a * (exp(-((x - x_0)^2 + (z - z_0)^2) / (2 * σ^2))) / (σ^2) *
+        Geometry.UWVector(x - x_0, z - z_0)
 
-        return div_uf
-    end
 
     n_elems_seq = 2 .^ (6, 7, 8, 9)
     err, Δh = zeros(length(n_elems_seq)), zeros(length(n_elems_seq))
 
     for (k, n) in enumerate(n_elems_seq)
         hv_center_space, hv_face_space =
             hvspace_2D(xlim = (-1, 1), zlim = (-1, 1), helem = n, velem = n)
+        ccoords = Fields.coordinate_field(hv_center_space)
+        bcoords = Fields.coordinate_field(hv_center_space.horizontal_space)
 
         Δh[k] = 1.0 / n
 
-        coords = Fields.coordinate_field(hv_center_space)
-        f = Fields.FieldVector(h = h_blob(coords, 0.5, 0.5))
+        h =
+            h_blob.(ccoords.x, ccoords.z, -0.5, -0.5) .*
+            Ref(Geometry.UWVector(1.0, 1.0))
+        bottom_flux =
+            h_blob.(bcoords.x, -1.0, -0.5, -0.5) .*
+            Ref(Geometry.UWVector(1.0, 1.0))
+        exact_divh =
+            Ref(Geometry.UWVector(1.0, 1.0)') .*
+            ∇h_blob.(ccoords.x, ccoords.z, -0.5, -0.5)
 
-        divuf = divergence(f, hv_center_space)
-        exact_divuf = div_h_blob(coords, 0.5, 0.5)
+        divh = divergence(h, bottom_flux)
 
-        err[k] = norm(exact_divuf .- divuf.h)
+        err[k] = norm(exact_divh .- divh)
     end
     # Divergence convergence rate
+    @test err[3] ≤ err[2] ≤ err[1] ≤ 0.01
     conv_comp_div = convergence_rate(err, Δh)
-    @test err[3] ≤ err[2] ≤ err[1] ≤ 0.1
+    #= not sure these are right?
```

Comment:
Can we confirm / update these tests?

Reply:
@jb-mackay what do these values give?

Reply:
```
julia> conv_comp_div = convergence_rate(err, Δh)
3-element Vector{Float64}:
 1.285293238893628
 0.7164519706218709
 0.5352264906826446
```

Reply:
Doesn't this look non-converging?

Reply:
Maybe we need a simpler example?

Reply:
Maybe base it on this one: https://github.com/CliMA/ClimaCore.jl/blob/main/test/Operators/finitedifference/column.jl#L687-L741
but make it spatially varying?